x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -53,11 +76,87 @@ public class IlmHealthIndicatorService implements HealthIndicatorService {
     );
 
     public static final String AUTOMATION_DISABLED_IMPACT_ID = "automation_disabled";
+    public static final String STAGNATING_INDEX_IMPACT_ID = "stagnating_index";
+    public static final List<HealthIndicatorImpact> AUTOMATION_DISABLED_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            AUTOMATION_DISABLED_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management is disabled. The performance and stability of the cluster "
+                + "could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    public static final List<HealthIndicatorImpact> STAGNATING_INDEX_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            STAGNATING_INDEX_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management cannot make progress on one or more indices. The performance and "
+                + "stability of the indices and/or the cluster could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    private static final TimeValue ONE_DAY = TimeValue.timeValueDays(1);
+
+    static final Map<String, RuleConfig> RULES_BY_ACTION_CONFIG = Map.of(
+        RolloverAction.NAME,
+        actionRule(RolloverAction.NAME).stepRules(stepRule(WaitForActiveShardsStep.NAME, ONE_DAY)),
+        //
+        MigrateAction.NAME,
+        actionRule(MigrateAction.NAME).maxTimeOnAction(ONE_DAY).noStepRules(),
+        //
+        SearchableSnapshotAction.NAME,
+        actionRule(SearchableSnapshotAction.NAME).maxTimeOnAction(ONE_DAY)
+            .stepRules(
+                stepRule(WaitForDataTierStep.NAME, ONE_DAY),
+                stepRule(WaitForIndexColorStep.NAME, ONE_DAY),
+                stepRule(WaitForNoFollowersStep.NAME, ONE_DAY)
+            ),
+        //
+        DeleteAction.NAME,
+        actionRule(DeleteAction.NAME).stepRules(stepRule(DeleteStep.NAME, ONE_DAY)),
+        //
+        ShrinkAction.NAME,
+        actionRule(ShrinkAction.NAME).maxTimeOnAction(ONE_DAY).stepRules(stepRule(WaitForNoFollowersStep.NAME, ONE_DAY)),
```

Comment:
Note (maybe in a comment) that we're adding this rule on the `WaitForNoFollowersStep` which runs on the leader node because the follower cluster will have an `unfollow` action injected before Shrink and SearchableSnapshot actions - the index should not be receiving writes anymore so the unfollow on the follower cluster should make sure that the `WaitForNoFollowersStep` on the leader is not taking too long.

We currently seem to not inject `Unfollow` before `Downsample` which I think is a bug for now, so please remove this `WaitForNoFollowersStep` rule from the `Downsample` action (and open a bug to fix and inject `Unfollow` before `Downsample`)